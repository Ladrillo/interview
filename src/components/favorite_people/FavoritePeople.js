import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../card/Card.js';


class FavoritePeople extends Component {
    componentWillMount() {
        this.props.dispatch({ type: 'PEOPLE/GET_FAVORITES' });
    }

    render() {
        const {
            favoritePeople,
            currentPage,
        } = this.props;

        return (
            <div>
            {
                favoritePeople.map(person => {
                    return (
                        <Card
                            key         = {person.id}
                            id          = {person.id}
                            name        = {person.name}
                            image       = {person.image}
                            birthday    = {person.birth_year}
                            homeworld   = {person.homeworldName}
                        />
                    );
                })
            }
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        currentPage:    state.currentPage,
        favoritePeople: state.favoritePeople,
    };
}

export default connect(mapStateToProps)(FavoritePeople);
