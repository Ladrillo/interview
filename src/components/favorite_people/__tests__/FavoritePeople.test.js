import FavoritePeople from '../FavoritePeople';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';


describe('FavoritePeople', () => {

    it('renders correctly', () => {
        const rendered = renderer.create(
            <FavoritePeople
                store = {{
                    getState:  () => ({ currentPage: 2, favoritePeople: [{id: 2}] }),
                    dispatch:  () => ({}),
                    subscribe: () => ({}),
                 }}
            />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
