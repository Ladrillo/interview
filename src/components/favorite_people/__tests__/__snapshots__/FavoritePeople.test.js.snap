exports[`FavoritePeople renders correctly 1`] = `
<div
  className="sc-htpNat lcAGFr"
  transform={true}>
  <div
    className="card">
    <div
      className="card-content">
      <div
        className="card-name">
         
      </div>
      <img
        alt="profile"
        src="http://localhost:3008/undefined" />
    </div>
  </div>
</div>
`;
